string name
string file
string command
